src/lib.rs

```diff
@@ -90,10 +90,10 @@ pub trait GraphWriter: Graph + GraphDataWriter {
 
 pub trait GraphDataWriter: Graph {
     /// Return a mutable reference to an edge's data, to annotate the edge.
-    fn edge_data_mut(&mut self, id: Id<Self::Edge>) -> Option<&mut Data<Self::Edge>>;
+    fn edge_data_mut(&mut self, id: &Id<Self::Edge>) -> Option<&mut Data<Self::Edge>>;
```

Comment:
Passing in a reference is fine I guess : ) - though I'm guessing our Ids won't be larger than a machine word and therefore will basically be free to clone/pass by value.